elmslie-core/src/main/kotlin/money/vivid/elmslie/core/store/Actor.kt

```diff
@@ -17,4 +26,21 @@ abstract class Actor<Command : Any, Event : Any> {
         errorMapper: (error: Throwable) -> Event? = { null },
     ) = mapNotNull { eventMapper(it) }
         .catch { errorMapper(it)?.let { event -> emit(event) } ?: throw it }
+
+    protected fun <T : Any, Command : Any> Flow<T>.asSwitchFlow(command: Command, delayMillis: Long = 0): Flow<T> {
+        return flow {
+            val switcher = mutex.withLock {
+                switchers.getOrPut(command::class) {
+                    Switcher()
+                }
+            }
+            switcher.switch(delayMillis) { this@asSwitchFlow }.collect {
+                emit(it)
+            }
+        }
+    }
+
+    protected fun <T : Any> cancelSwitchFlow(command: KClass<out Any>): Flow<T> {
+        return switchers[command]?.cancel() ?: emptyFlow()
```

Comment:
let's remove switch from map after cancel

Reply:
for what? we can reuse instance while actor is alive

Reply:
if we add custom keys to the switchers map, it would be possible to create an unlimited count of switchers. Of course, it's a theoretical issue, but still it's possible

I don't mind staying it as is, just want to highlight it

Reply:
according to the discussion below, it's not required